### PR TITLE
[CMM] Implement Sparkshaper Visionary

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Sparkshaper Visionary]]

**Comment by xenohedron** [comment]
Test failure appears to be just a typo in the new static filter.

Btw, we should probably only add new StaticFilters if they're actually used in several cards.

File: Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(5);
+
+        // At the beginning of combat on your turn, choose any number of target planeswalkers you control. Until end of turn, they become 3/3 blue Bird creatures with flying, hexproof, and "Whenever this creature deals combat damage to a player, scry 1."
+        TriggeredAbility triggeredAbility =
+            new BeginningOfCombatTriggeredAbility(
+                new BecomesCreatureTargetEffect(
+                    new SparkshaperVisionaryToken(),
+                    false, false, Duration.EndOfTurn,
+                    false, false, true
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java:46]: I believe this should be keeping abilities. Might want to add the reminder text as well.

Comment (Susucre) [reply Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java:46]: Oh I did miss that `keepAbilities` intent. `losaAllAbilities` to true was keeping the abilities in my test.

For the reminder text, sure I'll add it.

Comment (xenohedron) [reply Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java:46]: I'm not totally sure I'm following the constructors correctly, they're kind of messy. If you've tested and it works properly that's fine.

**Review by xenohedron — APPROVED** [review]

File: Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java (hunk 1 above)
Comment (Susucre) [reply Mage.Sets/src/mage/cards/s/SparkshaperVisionary.java]: This is the final values, I suspect the only difference would be for gaining types from static ability? really not sure.
becometarget

And it indeed works as far as I can tell. I had many Jaces living the Bird life (I did remove the scry reminder in the meantime)
 
jacebird

**Pull request merged**